### PR TITLE
Fix issues found by linter

### PR DESCRIPTION
**Pull request by p-strusiewiczsurmacki-mobica**
This PR is continuation of #36 
It fixes all the issues found by the golangci-lint configuration introduced in #36.

NOTE: This PR should be fully tested in production-grade test environment before being merged, as I am not able to do so.

### CODE REVIEW

File: pkg/reconciler/layer2.go [hunk 1]
```diff
@@ -1,61 +1,130 @@
...
+		bridgeID, err := r.netlinkManager.GetBridgeID(info)
+		if err != nil {
+			return fmt.Errorf("error getting bridge id for vlanId %d: %w", info.VlanID, err)
+		}
+		*anycastTrackerInterfaces = append(*anycastTrackerInterfaces, bridgeID)
+	}
+	return nil
+}
+
+func (r *reconcile) createDesired(l2vnis []networkv1alpha1.Layer2NetworkConfiguration) ([]nl.Layer2Information, error) {
```
Comment (schrej) [inline pkg/reconciler/layer2.go:124]: maybe call this `getDesired` or something like that. By prefixing everything with `create` it's not clear which functions actually create/apply configuration, and which just create/determine lists of configs.

File: pkg/reconciler/layer2.go [hunk 2]
```diff
@@ -81,79 +150,38 @@ func (r *reconcile) reconcileLayer2(l2vnis []networkv1alpha1.Layer2NetworkConfig
...
-	existing, err := r.netlinkManager.ListL2()
-	if err != nil {
-		return err
-	}
+	return desired, nil
+}
 
-	delete := []nl.Layer2Information{}
-	for _, cfg := range existing {
+func createToDelete(existing, desired []nl.Layer2Information) []nl.Layer2Information {
```
Comment (schrej) [inline pkg/reconciler/layer2.go:156]: Same here, the function name isn't great. Maybe something like `determineToBeDeleted()` or `findToBeDeleted()` instead?

File: pkg/nl/list.go [hunk 3]
```diff
@@ -1,149 +1,160 @@
...
 	if err != nil {
-		return err
+		return fmt.Errorf("error listing link's IPv6 addresses: %w", err)
 	}
-	for _, addr := range currentV4 {
+	for i, addr := range currentV4 {
 		if addr.Scope != unix.RT_SCOPE_UNIVERSE {
 			continue
 		}
-		info.AnycastGateways = append(info.AnycastGateways, &addr)
```
Comment (schrej) [inline pkg/nl/list.go:99]: @chdxD1 this was a bug that is getting fixed in this PR.
By simply referencing it this way, the same address is added multiple times: https://husni.dev/beware-of-implicit-memory-aliasing-in-go-foor-loop/

File: pkg/nl/list.go [hunk 4]
```diff
@@ -1,149 +1,160 @@
...
+		info.AnycastGateways = append(info.AnycastGateways, &currentV4[i])
 	}
-	for _, addr := range currentV6 {
+	for i, addr := range currentV6 {
 		if addr.Scope != unix.RT_SCOPE_UNIVERSE {
 			continue
 		}
-		info.AnycastGateways = append(info.AnycastGateways, &addr)
```
Comment (schrej) [inline pkg/nl/list.go:105]: @chdxD1 same here.

**Review by schrej — COMMENTED** [review]
Good job!

A few nits, but looks good overall.

I'll leave it to @chdxD1 to test this and to consider the two fixed allocation bugs.

File: pkg/anycast/anycast.go [hunk 5]
```diff
@@ -11,21 +11,22 @@ import (
...
+// TODO: Anycast Support is currently highly experimental.
 
-func (a *AnycastTracker) checkTrackedInterfaces() {
-	for _, intfIdx := range a.TrackedBridges {
+func (t *Tracker) checkTrackedInterfaces() {
+	for _, intfIdx := range t.TrackedBridges {
 		intf, err := netlink.LinkByIndex(intfIdx)
 		if err != nil {
-			fmt.Printf("Couldn't load interface idx %d: %v\n", intfIdx, err)
+			_, _ = fmt.Printf("Couldn't load interface idx %d: %v\n", intfIdx, err)
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/anycast/anycast.go:29]: Done

File: pkg/frr/configure.go [hunk 6]
```diff
@@ -42,78 +43,78 @@ func (m *FRRManager) Configure(in FRRConfiguration) (bool, error) {
...
 
-		return true, err
+		return true, nil
 	}
 	return false, nil
 }
 
-func (f *FRRManager) renderSubtemplates(in FRRConfiguration) (*FRRTemplateConfig, error) {
-	vrfRouterId, err := (&nl.NetlinkManager{}).GetUnderlayIP()
+func (*Manager) renderSubtemplates(in Configuration) (*templateConfig, error) {
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/frr/configure.go:56]: I've just made minimal changes to fix the issues, and as it was before a method it stayed this way. Anyway you;re right, changed that to be a function.

File: pkg/reconciler/reconciler.go [hunk 7]
```diff
@@ -30,32 +32,32 @@ type Reconciler struct {
...
 	if err := reconciler.frrManager.Init(); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error trying to init FRR Manager: %w", err)
 	}
 
-	if config, err := config.LoadConfig(); err != nil {
-		return nil, err
-	} else {
-		reconciler.config = config
+	var cfg *config.Config
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/reconciler/reconciler.go:55]: You're right, we don't. Fixed.

File: pkg/reconciler/layer2.go (hunk 2 above)
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/reconciler/layer2.go:156]: Done. Went with `determineToBeDeleted`.

File: pkg/bpf/loader.go [hunk 8]
```diff
@@ -161,7 +177,7 @@ func RunInterfaceCheck() {
 	go func() {
 		for {
 			checkTrackedInterfaces()
-			time.Sleep(5 * time.Second)
+			time.Sleep(sleepTime)
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/bpf/loader.go]: Done

File: pkg/config/config.go [hunk 9]
```diff
@@ -1,14 +1,15 @@
...
 	"os"
 
 	"gopkg.in/yaml.v2"
 )
 
 var (
-	VNI_MAP_FILE          = "/opt/network-operator/config.yaml"
-	SKIP_VRF_TEMPLATE_VNI = -1
+	vniMapFile         = "/opt/network-operator/config.yaml"
+	SkipVrfTemplateVni = -1
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/config/config.go:12]: Yes, it's used in other packages. But changed those 2 vars to be constants.

File: pkg/macvlan/macvlan.go [hunk 10]
```diff
@@ -155,7 +170,7 @@ func RunMACSync(interfacePrefix string) {
 		go func() {
 			for {
 				checkTrackedInterfaces()
-				time.Sleep(5 * time.Second)
+				time.Sleep(defaultSleep)
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/macvlan/macvlan.go:173]: Done

File: pkg/nl/layer2.go [hunk 11]
```diff
@@ -29,53 +34,66 @@ type Layer2Information struct {
...
 	}
 	info.bridge = bridge
 	if err := n.configureBridge(bridge.Name); err != nil {
-		return err
+		return nil, err
 	}
 
 	// Wait 500ms before configuring anycast gateways on newly added interface
-	time.Sleep(500 * time.Millisecond)
+	time.Sleep(waitTimeInMs * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/nl/layer2.go:85]: Done

File: pkg/notrack/notrack.go [hunk 12]
```diff
@@ -69,64 +71,66 @@ func reconcileIPTables(notrackLinks []string, ipt *iptables.IPTables) error {
...
-			time.Sleep(20 * time.Second)
+			if err := ipt4.AppendUnique(iptablesTable, iptablesOutput, "-s", underlayIP.String(), "-p", "udp", "--dport", "4789", "-j", "NOTRACK"); err != nil {
+				notrackLog.Error(err, "error reconciling VXLAN notrack in IPv4 iptables")
+			}
+		} else {
+			notrackLog.Error(err, "error reconciling VXLAN notrack in IPv4 iptables")
 		}
-	}()
+
+		time.Sleep(sleepTime)
```
[the remark this answers is not in the document]
Comment (p-strusiewiczsurmacki-mobica) [reply pkg/notrack/notrack.go:134]: Done

**Pull request merged**